pkg/config/seelog_config.go

```diff
@@ -0,0 +1,92 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package config
+
+import (
+	"bytes"
+	"strings"
+	"sync"
+	"text/template"
+)
+
+// SeelogConfig abstract seelog XML configuration definition
+type SeelogConfig struct {
+	settings map[string]interface{}
+	m        sync.Mutex
```

Comment:
That's a nit but you can use composition here, the `Lock()` and `Unlock()` will then be callable with a nicer syntax, example with configureSyslog:

```
func (c *SeelogConfig) configureSyslog(syslogURI string, usetTLS bool) {
	c.Lock()
	defer c.Unlock()
	c.settings["syslogURI"] = syslogURI
	c.settings["syslogUseTLS"] = usetTLS
}
```

Reply:
It makes sense and the result is easier to read. Will do.